navigation.py

```diff
@@ -0,0 +1,237 @@
+        dropdown_id -- the id of the dropdown menu
+        option_text -- the text of the dropdown item to be selected
+    """
+    dropdown_menu = Select(driver.find_element_by_id(dropdown_id))
+    dropdown_menu.select_by_visible_text(option_text)
+
+def check_checkbox(driver, box_id, should_end_up_selected = True):
+    """Interacts with a checkbox, will tick the box by default.
+
+    Positional arguments:
+        driver -- the Selenium driver to use
+        box_id -- the id of the checkbox
+    Optional arguments:
+        should_end_up_selected -- box will end up ticked if True, else uticked. Default True.
+    """
+    checkbox = driver.find_element_by_id(box_id)
+    if checkbox.get_attribute("type") != "checkbox":
+        raise UnexpectedLayoutError(f"The element with id {box_id} is not a checkbox.")
+    if checkbox.is_selected() != should_end_up_selected:
+        checkbox.click()
+
+
+def wait_for_next_page(driver):
+    # wait for driver to start loading next page
+    time.sleep(0.5)
+    # wait for page to load, up to ten seconds
+    WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.XPATH, "//html")))
+
+
+def try_login(driver, username, password):
+    navigate_to_page("login", driver)
+
+    fill_form_field(username, driver, "input_username")
+    fill_form_field(password, driver, "input_password")
+    submit_form(driver)
+
+    wait_for_next_page(driver)
+
+def request_password_reset(driver, user_email):
+    driver.get(pages["login"])
+
+    forgot_password_link = driver.find_element_by_xpath("//p[@class='forgot']/a")
+    forgot_password_link.click()
+
+    WebDriverWait(driver, 10).until(EC.url_changes(pages["login"]))
+
+    email_entry = driver.find_element_by_id("input_username")
+    email_entry.send_keys(user_email)
+    email_entry.submit()
+
+    WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.ID, "input_reset-token")))
+
+    # wait for token to be written to database
+    time.sleep(0.5) 
+
+def enter_password_request_token(driver, reset_token):
+    WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.ID, "input_reset-token")))
+
+    reset_token_entry = driver.find_element_by_id("input_reset-token")
+    reset_token_entry.send_keys(reset_token)
+    reset_token_entry.submit()
+
+    WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.ID, "input_new_password")))
+
+def enter_new_password_for_reset(driver, new_password):
+    WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.ID, "input_new_password")))
+    password_entry = driver.find_element_by_id("input_new_password")
+    password_entry.send_keys(new_password)
+
+    password_confirm = driver.find_element_by_id("input_confirm_password")
+    password_confirm.send_keys(new_password)
+    password_confirm.submit()
+
+    WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.ID, "input_username")))
+
+if __name__ == "__main__":
```

Comment:
Do you think we need to keep this? I would prefer to remove it (and only have an `if __name__ == "__main__"` if it makes sense to run the python file as a standalone)